src/main/java/me/magnet/consultant/RoutingStrategies.java

```diff
@@ -0,0 +1,110 @@
+package me.magnet.consultant;
+
+import java.util.List;
+import java.util.Random;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Multimap;
+import com.google.common.collect.Sets;
+
+public class RoutingStrategies {
+
+	public static final RoutingStrategy RANDOMIZED_WEIGHTED_DISTANCE = new RoutingStrategy() {
+
+		private final Random random = new Random();
+
+		@Override
+		public List<ServiceInstance> listInstances(ServiceLocator serviceLocator, String serviceName) {
+			List<ServiceInstance> instances = serviceLocator.listInstances(serviceName);
+			if (instances.size() <= 1) {
+				// No routing to do.
+				return instances;
+			}
+
+			// Pick a random number between 0-1.
+			double chance = random.nextDouble();
+
+			/*
+			 * Pick the index of the first instance to try based on this chance.
+			 * If the chance was < 0.5 then use the instance on index 0.
+			 * If the chance was >= 0.5 and < 0.75 then use the instance on index 1.
+			 * If the chance was >= 0.75 and < 0.875 then use the instance on index 2.
+			 * If the chance was >= 0.875 and < 0.9375 then use the instance on index 3.
+			 * Etc...
+			 */
+			int index = 0;
+			double threshold = 0.5;
+			while (threshold < chance && index < instances.size() - 1) {
+				index++;
+				threshold = Math.pow(0.5, index + 1);
+			}
```

Comment:
I think that this function doesn't work as advertised. 

- If `chance < 0.5` then the loop will be skipped completely (due to first condition) -> `index = 0`
- If `chance > 0.5` then the loop will roll until the end, ending up with `index = instances.size() - 1`: every threshold value will be below 0.5, since `0.5^n < 0.5` for `n >= 1`. This means that the loop can only be exited when `index >= instances.size() - 1`.

So there's a 50% chance the first instance will be used, and a 50% chance the last instance will be used.

I think line 42 should be `threshold = 1 - Math.pow(0.5, index + 1)` instead.